File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        x = []
        for i in range(0,len(nums1)):
            
            for j in range(0,len(nums2)):
            
                f= 1
                v=0
                if nums1[i]==nums2[j]:
                    v = nums2[j]
        
                    for k in range(j+1,len(nums2)):
                        if nums2[k]>v:
                            x.append(nums2[k])
                            f= 0
                            break
                    
                    if f== 1:
                        x.append(-1)
        return x

File: 0496-next-greater-element-i/0496-next-greater-element-i.java
import java.util.Stack;
import java.util.HashMap;
import java.util.Map;
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        Stack<Integer> x=new Stack<>();
        Map<Integer,Integer> y=new HashMap<>();
        
        for(int i:nums2){
        while(!x.isEmpty() && i>x.peek()){
            y.put(x.pop(),i);
            
            
        }
        x.push(i);    
        }
while(!x.isEmpty()){
    y.put(x.pop(),-1);
}
        int q[]=new int[nums1.length];
        for(int i=0;i<nums1.length;i++){
            q[i]=y.get(nums1[i]);
            
        }
return q;
    }
}